core/b2b/map-to-b2b-product-options.tsx

```diff
@@ -0,0 +1,63 @@
+import { B2BProductOption } from '~/b2b/types';
+
+import { Field } from '../vibes/soul/sections/product-detail/schema';
+
+interface ProductOption {
+  field: Field;
+  value?: string | number;
+}
+
+export function mapToB2BProductOptions({ field, value }: ProductOption): B2BProductOption {
+  const fieldId = Number(field.name);
+
+  const baseOption: B2BProductOption = {
+    optionEntityId: fieldId,
+    optionValueEntityId: 0, // Will be set based on type
+    entityId: fieldId,
+    valueEntityId: 0, // Will be set based on type
+    text: '',
+    number: 0,
+    date: { utc: '' },
+  };
+
+  switch (field.type) {
+    case 'text':
+    case 'textarea':
+      return {
+        ...baseOption,
+        text: String(value || ''),
```

Comment:
Seems weird that we destructure all values in `baseOption`, like number and date, for a field of type `text`. Does the API require all fields at all times? Or are we missing some schema types here?

Reply:
we are indeed missing types here, but the problem originates at the buyer portal, and it is not an easy fix at this point in time. Something to keep in mind if/when we rewrite this piece.